api/src/test/java/org/openmrs/module/sync/SyncProgramAttributeTest.java

```diff
@@ -127,7 +128,46 @@ public void runOnParent() throws Exception {
 			}
 		});
 	}
-	
+
+	@Test
+	@Transactional(propagation = Propagation.NOT_SUPPORTED)
+	public void shouldSyncProgramAttributeWhichWasDeleted() throws Exception {
+		runSyncTest(new SyncTestHelper() {
+
+			public void runOnChild() throws Exception {
+				ProgramAttributeType programAttributeType = programService.getProgramAttributeTypeByUuid("42ec23bf-b9f9-11ed-9873-0242ac120002");
+				Assertions.assertNotNull(programAttributeType);
+				Assertions.assertEquals(programAttributeType.getName(), "Transfer Status");
+				PatientProgram pp = programService.getPatientProgram(2);
+				List<PatientProgramAttribute> activeAttributes = pp.getActiveAttributes(programAttributeType);
+				Assertions.assertEquals(1, activeAttributes.size());
+				PatientProgramAttribute attr = activeAttributes.get(0);
+				Assertions.assertEquals(attr.getValue(), "transfer-in");
+				//activeAttributes.remove(0); -- it does not remove the attribute
+				//pp.getAttributes().remove(0);
+				Set<PatientProgramAttribute> ppAttributes = pp.getAttributes();
+				Iterator<PatientProgramAttribute> iterator = ppAttributes.iterator();
+				while(iterator.hasNext()){
+					iterator.remove();
```

Comment:
@mseaton , I have made a few different unsuccessful attempts here to remove a patient_program_attribute. I am not sure if my syntax is not correct, or the OpenMRS API does not yet support this method? This last attempt that is not commented out, iterator.remove() throws IllegalStateException. Can you please have a look at this code, and let me know if there is a better way to do this? Thanks!